test/unit/fact_value_test.rb

```diff
@@ -122,22 +122,22 @@ def setup
     end
 
     test "only return facts from host in user's taxonomies" do
-      user_role = FactoryGirl.create(:user_user_role)
-      FactoryGirl.create(:filter, :role => user_role.role, :permissions => Permission.where(:name => 'view_hosts'), :search => "hostgroup_id = #{target_host.hostgroup_id}")
+      setup_user('view', 'hosts', "hostgroup_id = #{target_host.hostgroup_id}")
 
       orgs = FactoryGirl.create_pair(:organization)
       locs = FactoryGirl.create_pair(:location)
-      target_host.update_attributes(:location => locs.last, :organization => orgs.last)
-
-      user_role.owner.update_attributes(:locations => [locs.first], :organizations => [orgs.first])
-      as_user user_role.owner do
-        assert_equal [], FactValue.my_facts.map(&:id).sort
+      as_admin do
+        [orgs, locs].map { |taxonomy| taxonomy.map { |t| t.update_attributes(:users => []) } }
```

Comment:
Apparently not, it's a bit messed up because users(:one) will show up in all locs/orgs. I'll do this changing the user from a fixture to FactoryGirl.